### PR TITLE
Add internal scope descriptions

### PR DESCRIPTION
**Pull request by PasinduYeshan**
### Proposed changes in this pull request
$subject

### Related Issues
- https://github.com/wso2/product-is/issues/18320

### CODE REVIEW

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml [hunk 1]
```diff
@@ -291,70 +345,81 @@
...
         <Scopes>
-            <Scope displayName="View Notification Senders" name="internal_notification_senders_view"/>
-            <Scope displayName="Create Notification Senders"
-                   name="internal_notification_senders_create"/>
-            <Scope displayName="Update Notification Senders"
-                   name="internal_notification_senders_update"/>
-            <Scope displayName="Delete Notification Senders"
-                   name="internal_notification_senders_delete"/>
+            <Scope displayName="View Notification Senders" name="internal_org_notification_senders_view" 
+                    description="View new notification senders for organization"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml:380]: based on how the description has been formed for other APIs, we shouldn't have `new` for view, update and delete descriptions right?

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml [hunk 2]
```diff
@@ -291,70 +345,81 @@
...
-            <Scope displayName="Update Organization Discovery"
-                   name="internal_organization_discovery_update"/>
-            <Scope displayName="Delete Organization Discovery"
-                   name="internal_organization_discovery_delete"/>
+            <Scope displayName="View Organization Discovery" name="internal_organization_discovery_view" 
+                    description="View discovery attributes of the organization (root)"/>
+            <Scope displayName="Update Organization Discovery" name="internal_organization_discovery_update" 
+                    description="Update discovery attributes of the organization (root)"/>
+            <Scope displayName="Delete Organization Discovery" name="internal_organization_discovery_delete" 
+                    description="Delete discovery attributes of the organization (root)"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml:422]: this API can view, update, delete discovery attributes of any organization within the root organization. Therefore the description should be updated to convey that meaning

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml [hunk 3]
```diff
@@ -526,467 +639,575 @@
...
                  requiresAuthorization="true" description="API representation of the Code Management API" type="TENANT_USER">
         <Scopes>
-            <Scope displayName="View Code" name="internal_code_mgt_view"/>
-            <Scope displayName="Create Code" name="internal_code_mgt_create"/>
-            <Scope displayName="Update Code" name="internal_code_mgt_update"/>
-            <Scope displayName="Delete Code" name="internal_code_mgt_delete"/>
+            <Scope displayName="View Code" name="internal_code_mgt_view" 
+                    description="View code in the organization (root)"/>
+            <Scope displayName="Create Code" name="internal_code_mgt_create" 
+                    description="Crate new code in the organization (root)"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml:770]:
```suggestion
                    description="Create new code in the organization (root)"/>
```

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml [hunk 4]
```diff
@@ -526,467 +639,575 @@
...
-            <Scope displayName="View Recovery" name="internal_org_recovery_view"/>
-            <Scope displayName="Create Recovery" name="internal_org_recovery_create"/>
-            <Scope displayName="Update Recovery" name="internal_org_recovery_update"/>
-            <Scope displayName="Delete Recovery" name="internal_org_recovery_delete"/>
+            <Scope displayName="View Recovery" name="internal_org_recovery_view" 
+                    description="View recovery in the organization"/>
+            <Scope displayName="Create Recovery" name="internal_org_recovery_create" 
+                    description="Create new recovery in  the organization"/>
+            <Scope displayName="Update Recovery" name="internal_org_recovery_update" 
+                    description="Update recovery in  the organization"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml:844]:
```suggestion
                    description="Update recovery in the organization"/>
```

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml [hunk 5]
```diff
@@ -526,467 +639,575 @@
...
+            <Scope displayName="Delete Recovery" name="internal_org_recovery_delete" 
+                    description="Delete recovery in  the organization"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml:846]:
```suggestion
                    description="Delete recovery in the organization"/>
```

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml [hunk 6]
```diff
@@ -526,467 +639,575 @@
...
+            <Scope displayName="Update User" name="internal_org_user_update" 
+                    description="Update user in the organization"/>
         </Scopes>
     </APIResource>
     <APIResource name="Me API" identifier="/me/" requiresAuthorization="false"
                  description="API representation of the Me API" type="OTHER">
         <Scopes>
-            <Scope displayName="User Login" name="internal_login"/>
+            <Scope displayName="User Login" name="internal_login" 
+                    description="Manage the your user account"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml:987]: typo

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2 [hunk 7]
```diff
@@ -300,70 +354,81 @@
...
         <Scopes>
-            <Scope displayName="View Notification Senders" name="internal_notification_senders_view"/>
-            <Scope displayName="Create Notification Senders"
-                   name="internal_notification_senders_create"/>
-            <Scope displayName="Update Notification Senders"
-                   name="internal_notification_senders_update"/>
-            <Scope displayName="Delete Notification Senders"
-                   name="internal_notification_senders_delete"/>
+            <Scope displayName="View Notification Senders" name="internal_notification_senders_view" 
+                    description="View new notification senders for the organization (root)"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2:375]:
```suggestion
                    description="View notification senders for the organization (root)"/>
```

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2 [hunk 8]
```diff
@@ -300,70 +354,81 @@
...
+            <Scope displayName="Create Notification Senders" name="internal_notification_senders_create" 
+                    description="Create notification senders for the organization (root)"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2:377]:
```suggestion
                    description="Create new notification senders for the organization (root)"/>
```

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2 [hunk 9]
```diff
@@ -300,70 +354,81 @@
...
-            <Scope displayName="Delete Notification Senders"
-                   name="internal_org_notification_senders_delete"/>
+            <Scope displayName="View Notification Senders" name="internal_org_notification_senders_view" 
+                    description="View new notification senders for organization"/>
+            <Scope displayName="Create Notification Senders" name="internal_org_notification_senders_create" 
+                    description="Create new notification senders for organization"/>
+            <Scope displayName="Update Notification Senders" name="internal_org_notification_senders_update" 
+                    description="Update new notification senders for organization"/>
+            <Scope displayName="Delete Notification Senders" name="internal_org_notification_senders_delete" 
+                    description="Delete new notification senders for organization"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2:395]: `new` should be removed for view, delete and update

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml (hunk 1 above)
Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml]: Removed

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml [hunk 10]
```diff
@@ -291,70 +345,81 @@
...
+                    description="Delete OAuth2 scopes and bindings"/>
         </Scopes>
     </APIResource>
     <APIResource name="Tenant Management API" identifier="/api/server/v1/tenants"
                  requiresAuthorization="true" description="API representation of the Tenant Management API" type="SYSTEM">
         <Scopes>
-            <Scope displayName="List Tenants" name="internal_list_tenants"/>
-            <Scope displayName="Modify Tenants" name="internal_modify_tenants"/>
+            <Scope displayName="List Tenants" name="internal_list_tenants" 
+                    description="Create/retrieve the organizations (root) and check domain availability"/>
```
[the remark this answers is not in the document]
Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml:407]: Yes same scope is used for GET, and POST.

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml (hunk 2 above)
Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml]: Fixed

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml (hunk 3 above)
Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml]: Fixed

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml (hunk 5 above)
Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml]: Removed

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml (hunk 4 above)
Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml]: Removed

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml (hunk 6 above)
Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml]: Fixed

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2 (hunk 8 above)
Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2]: Removed

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2 (hunk 9 above)
Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml.j2]: Removed

File: features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml [hunk 11]
```diff
@@ -415,11 +415,11 @@
...
         <Scopes>
             <Scope displayName="View Organization Discovery" name="internal_organization_discovery_view" 
-                    description="View discovery attributes of the organization (root)"/>
+                    description="View discovery attributes of organizations within the root organization"/>
             <Scope displayName="Update Organization Discovery" name="internal_organization_discovery_update" 
-                    description="Update discovery attributes of the organization (root)"/>
+                    description="Update discovery attributes of organizations within the root organization"/>
             <Scope displayName="Delete Organization Discovery" name="internal_organization_discovery_delete" 
-                    description="Delete discovery attributes of the organization (root)"/>
+                    description="Delete discovery attributes of organizations within the root organization"/>
```
Comment (dewniMW) [inline features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml:422]: within -> under

Comment (PasinduYeshan) [reply features/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt.server.feature/resources/system-api-resource.xml]: Removed

**Review by dewniMW — APPROVED** [review]

**Pull request merged**